Libplanet/Blockchain/BlockChain.Swap.cs

```diff
@@ -0,0 +1,462 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Libplanet.Action;
+using Libplanet.Blockchain.Renderers;
+using Libplanet.Blocks;
+using Libplanet.Store;
+using Libplanet.Tx;
+
+namespace Libplanet.Blockchain
+{
+    public partial class BlockChain<T>
+    {
+        // FIXME it's very dangerous because replacing Id means
+        // ALL blocks (referenced by MineBlock(), etc.) will be changed.
+        // we need to add a synchronization mechanism to handle this correctly.
+        internal System.Action Swap(
+            BlockChain<T> other,
+            bool render,
```

Comment:
IMO, If the rendering is done directly by calling the returned function, this flag seems to be unnecessary now...

Reply:
I wasn't sure what the inline comment was referring to, i.e. for which case a synchronization mechanism is needed.

Reply:
Rendering may be okay from the synchronization issue. (When I first wrote that, I was worried because of the consistency of data between chains.)

Reply:
I meant, `render` flag doesn't seem to be needed anymore. (not a comment 😅)
At least as far as I understand, code that calls Swap() can now decide whether to render directly without going through the render flag. That is, in the past

```csharp
bool render = IsRenderNeeded();
chain.Swap(render: render);
```

but we can use after this PR like...

```csharp
Action renderSwap = chain.Swap();
if (IsRenderNeeded()) {
    renderSwap();
}
```

Please let me know if I'm misunderstanding anything.

Reply:
Ah... sorry about that. I misunderstood the original comment then. However, I agree with the rationale, but as in the comment below, I'd like to deal with it in another PR if possible.

Reply:
🆗  fair enough.